geomstats/geometry/positive_lower_triangular_matrices.py

```diff
@@ -0,0 +1,359 @@
+
+        Parameters
+        ----------
+        point : array-like, shape=[..., n, n]
+            Matrix to project.
+
+        Returns
+        -------
+        projected: array-like, shape=[..., n, n]
+            SPD matrix.
+        """
+        vec_diag = gs.abs(Matrices.diagonal(point) - 0.1) + 0.1
+        diag = gs.vec_to_diag(vec_diag)
+        strictly_lower_triangular = Matrices.to_lower_triangular(point)
+        projection = diag + strictly_lower_triangular
+        return projection
+
+    @staticmethod
+    def gram(point):
+        """Compute gram matrix of rows
+
+        Gram_matrix is mapping from point to point.point^{T}.
+        This is diffeomorphism between cholesky space and spd manifold
+
+        Parameters
+        ----------
+        point : array-like, shape=[..., n, n]
+            element in cholesky space
+
+        Returns
+        -------
+        projected: array-like, shape=[..., n, n]
+            SPD matrix.
+        """
+        return gs.einsum("...ij,...kj->...ik", point, point)
+
+    @staticmethod
+    def differential_gram(tanget_vec, base_point):
+        """Compute gram matrix of rows
+
+        Gram_matrix is mapping from point to point.point^{T}.
+        This is diffeomorphism between cholesky space and spd manifold
+
+        Parameters
+        ----------
+        tangent_vec : array_like, shape=[..., n, n]
+            Tangent vector at base point.
+        base_point : array_like, shape=[..., n, n]
+            Base point.
+
+        Returns
+        -------
+        differential_gram : array-like, shape=[..., n, n]
+            Differential of the matrix exponential.
+        """
+        mat1 = gs.einsum("...ij,...kj->...ik", tanget_vec, base_point)
+        mat2 = gs.einsum("...ij,...kj->...ik", base_point, tanget_vec)
+        return mat1 + mat2
+
+    @staticmethod
+    def inverse_differential_gram(tanget_vec, base_point):
+        """Compute inverse differential of gram map
+
+        Parameters
+        ----------
+        tangent_vec : array_like, shape=[..., n, n]
+            tanget vector at gram(base_point)
+            Symmetric Matrix
+        base_point : array_like, shape=[..., n, n]
+            Base point.
+
+        Returns
+        -------
+        inverse_differential_gram : array-like, shape=[..., n, n]
+            Inverse differential of gram.
+            Lower triangular matrix
+        """
+        inv_base_point = gs.linalg.inv(base_point)
+        inv_transpose_base_point = Matrices.transpose(inv_base_point)
+        aux = Matrices.to_lower_triangular_diagonal_scaled(
+            Matrices.mul(inv_base_point, tanget_vec, inv_transpose_base_point)
+        )
+        inverse_differential_gram = Matrices.mul(base_point, aux)
+        return inverse_differential_gram
+
+
+class CholeskyMetric(RiemannianMetric):
+    """Class for the cholesky metric on the cholesky space."""
+
+    def __init__(self, n):
+        """Build the CholeskyMetric
```

Comment:
Init do not have docstrings: the description of the parameters and the reference should go in the docstring of the class.